contracts/BnomialNFT.sol

```diff
@@ -24,12 +25,15 @@ contract BnomialNFT is ERC721, Ownable {
         return _tokenIdCounter.current();
     }
 
-    function mint(address to, uint256 badge) external onlyOwner {
+    function mint(address to) external {
         require(balanceOf(to) == 0, "Only one token per wallet allowed");
+        require(_badges[to].length > 0, "At least one achievement is needed");
+        _tokenIdCounter.increment();      
+        _safeMint(to, _tokenIdCounter.current());       
+    }
 
-        _tokenIdCounter.increment();
-        _safeMint(to, _tokenIdCounter.current());
-        addBadge(to, badge);
+    function canMint(address owner) public view returns (bool) {
+        return (_badges[owner].length > 0) ? true : false;
```

Comment:
I think this can be simplified as (not 100% sure it works in Solidity, but should work in C++/Python):
```
return (_badges[owner].length > 0);
```

Reply:
yes, it works very well